Cabal/src/Distribution/Simple/Program/Db.hs

```diff
@@ -339,7 +339,7 @@ configureProgram verbosity prog progdb = do
       if absolute
         then return (Just (UserSpecified path, []))
         else findProgramOnSearchPath verbosity (progSearchPath progdb) path
-             >>= maybe (die' verbosity notFound)
+             >>= maybe (dieAll verbosity notFound (Just path) 77)
```

Comment:
`path` goes to stdout and we set exit code to 77... this can serve as a hint for calling tools, which may parse stdout and do stuff with it

Reply:
If one doesn't like stdout (because bad scripts don't check exit code before eating stdout): another approach might be to have a better format for the stderr message, that can be parsed.

Reply:
I think it might be reasonable to guard this behavior through a cli switch. It might break existing scripts.

Reply:
True, this is a breaking change, otherwise. Additionally, I don't think we can in general rely on that nothing else has written to stdout, since e.g. haddock and hpc just pollute everything.

Reply:
> I don't think we can in general rely on that nothing else has written to stdout, since e.g. haddock and hpc just pollute everything

I'm not sure I follow. What do you mean?

Reply:
Just a general concern that stdout is often polluted by other external tools, such as haddock and hpc. In this specific situation, it should be safe for external tools, such as hls, to try to read the last line of stdout to get the path.

Reply:
I don't think there will be any stdout when `cabal exec ghc` fails to find `ghc`.